sdks/python/apache_beam/internal/apiclient.py

```diff
@@ -225,11 +226,18 @@ def __init__(self, packages, options, environment_version):
       self.proto.sdkPipelineOptions = (
           dataflow.Environment.SdkPipelineOptionsValue())
 
-      for k, v in sdk_pipeline_options.iteritems():
-        if v is not None:
-          self.proto.sdkPipelineOptions.additionalProperties.append(
-              dataflow.Environment.SdkPipelineOptionsValue.AdditionalProperty(
-                  key=k, value=to_json_value(v)))
+      options_dict = {k: v
+                      for k, v in sdk_pipeline_options.iteritems()
+                      if v is not None}
+      self.proto.sdkPipelineOptions.additionalProperties.append(
+          dataflow.Environment.SdkPipelineOptionsValue.AdditionalProperty(
+              key='options', value=to_json_value(options_dict)))
+
+      dd = DisplayData.create_from(options)
+      items = [item.get_dict() for item in dd.items]
+      self.proto.sdkPipelineOptions.additionalProperties.append(
+          dataflow.Environment.SdkPipelineOptionsValue.AdditionalProperty(
+              key='display_data', value=to_json_value(items)))
```

Comment:
@chamikaramj another question: Do you approve of the way in which I am adding display_data to the options proto? : )